src/compiler/checker.ts

```diff
@@ -38321,6 +38321,9 @@ namespace ts {
             }
             // For a binding pattern, validate the initializer and exit
             if (isBindingPattern(node.name)) {
+                if (findAncestor(node, n => isInterfaceDeclaration(n) || isTypeLiteralNode(n) || isTypeAliasDeclaration(n) || !!(n.flags & NodeFlags.Ambient))) {

```

Comment:
The check is loosely~ based on this one [here](https://github.dev/microsoft/TypeScript/blob/a8e13f7340c5229426072d443fa511bba82a0054/src/compiler/checker.ts#L1656-L1658). I think that this other one is incomplete - and I'm also working on fixing it.

Reply:
Yeah, it might be worth pulling out that helper and fixing it.

Reply:
Here is the fix: https://github.com/microsoft/TypeScript/pull/50824 . Once one of those PR lands I can rebase the other one to unify this helper

Reply:
Thanks for sending it!